### PR TITLE
[Distributed] Distributed METIS partition

### PR DESCRIPTION
**Pull request by zheng-da**
## Description
This script is to convert METIS partitions into DGLGraphs.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/pytorch/rgcn/experimental/README.md [hunk 1]
```diff
@@ -88,6 +88,58 @@ We can get the performance score at the second epoch:
...
+More details about the four steps are explained in our
+[user guide](https://doc.dgl.ai/guide/distributed-preprocessing.html).
+
+### Step 1: write the graph into files.
+
+The graph structure should be written as a node file and an edge file. The node features and edge features
+can be written as DGL tensors. `write_mag.py` shows an example of writing the OGB MAG graph into files.
+
+```bash
+python3 write_mag.py
```
Comment (zhjwy9343) [inline examples/pytorch/rgcn/experimental/README.md:103]: Run the command will get an error:

Traceback (most recent call last):
  File "write_mag.py", line 42, in <module>
    node_data = th.stack(node_data, 1)
RuntimeError: Expected object of scalar type long int but got scalar type bool for sequence element 1.

File: tools/convert_partition.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
+import os
+import json
+import time
+import argparse
+import numpy as np
+import dgl
+import torch as th
+import pyarrow
+from pyarrow import csv
```
Comment (jermainewang) [inline tools/convert_partition.py:9]: Remember to add package dependencies in the README.

**Comment by zhjwy9343** [comment]
I tested the rgcn/experimental codes about parMETIS. In the 2nd round, all python scripts pasted tests.

In addition, have two requirements:
1. For users who do not know what "weights" mean or do not care about the performance of different weights, but just want to run distributed partitioning, we can allow them not setting weight. If no weights, DGL partition code can have a default weights. If our users care about performance of using weights, they can set weights for nodes and edges, overwriting the default setting in DGL.
2. The convert_partition.py should be packed into DGL, otherwise the relative path used in Step 3 will fail if users use pip to install DGL.

**Review by zhjwy9343 — COMMENTED** [review]
See my comments in general.

File: tools/convert_partition.py (hunk 2 above)
Comment (zheng-da) [reply tools/convert_partition.py:9]: in which README?

**Comment by zheng-da** [comment]
For the first comment, I need to wait for George to change ParMETIS.

For the second comment, do we still have time to pack the script into DGL? @jermainewang We should also pack the launch script to DGL.

**Review by jermainewang — APPROVED** [review]

**Pull request merged**